code/src/components/ScanBarcode.js

```diff
@@ -0,0 +1,141 @@
+import React, { useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux';
+import { BarcodeScanner } from 'components/BarcodeScanner'
+import { getProduct, barcodeReducer } from 'reducers/barcodeReducer'
+import { ui } from 'reducers/ui'
+import { ShowProduct } from 'components/ShowProduct';
+import { ListSaved } from 'components/ListSaved';
+import { NotFound } from 'components/NotFound';
+import { Button, ButtonBracket } from 'lib/Buttons';
+import { BarcodeAnimation } from 'lib/BarcodeAnimation';
+import styled from 'styled-components';
+
+export const ScanBarcode = () => {
+  const [showScanner, setShowScanner] = useState(false)
+  const product = useSelector(store => store.reducer.product)
+  const productsArray = useSelector(store => store.reducer.productsArray)
+
+  const dispatch = useDispatch();
+
+  const onDetected = (code) => {
+    setShowScanner(false);
+    if (code > 1000000) {
+      if (product && product.id === code) { // Don't fetch same product twice
+        console.log('Code same as current product');
+        dispatch(ui.actions.setNotFound(false))
+        return;
+      }
+      if (productsArray.length > 0 && productsArray.find(product => product.id === code)) {
+        console.log('Code in saved products');
+        dispatch(ui.actions.setNotFound(false))
+        dispatch(barcodeReducer.actions.setProduct(productsArray.find(product => product.id === code)))
+        return;
+      }
+      console.log(`New code: ${code}`);
+      dispatch(getProduct(code));
+    }
+  };
+
+  return (
+    <Wrapper>
+      <Code>
+        {!showScanner && (
+          <Button type="button" onClick={() => setShowScanner(true)}>
+            Start scanner
+          </Button>
+        )}
+        <Manual>
+          <label>
+            Or enter barcode:{' '}
+            <Input type="text" onChange={(e) => onDetected(e.target.value)} />
+          </label>
+          <Links>
+            <ButtonBracket onClick={() => onDetected('7311070347272')}>Giflar</ButtonBracket>
+            <ButtonBracket onClick={() => onDetected('7300156573186')}>Milk</ButtonBracket>
+            <ButtonBracket onClick={() => onDetected('7318693440007')}>Potatos</ButtonBracket>
+          </Links>
+        </Manual>
+      </Code>
+
+      {showScanner && (
+        <BarcodeScanner onDetected={onDetected} />
+      )}
+
+      <NotFound />
+      {!product && !showScanner && (<BarcodeAnimation />)}
+      {product && (<ShowProduct />)}
+      {productsArray.length > 0 && (<ListSaved />)}
```

Comment:
Why did you decide to mount these in ScanBarcode.js instead of in App.js?

Reply:
Hm, I think it was because I can't access the useSelector there for the conditional rendering. But I guess I could move the condition inside the component, to get a cleaner code over all?